### PR TITLE
Car manager

### PR DESCRIPTION
**Pull request by Konrad-Rys**
Goal of this story is to create api for car managing. Please review

### CODE REVIEW

File: src/main/java/com/guysfromusa/carsgame/services/CarService.java [hunk 1]
```diff
@@ -1,55 +1,21 @@
...
-        Direction newDirection = turnF.apply(car.getDirection());
-        car.setDirection(newDirection);
-        */
-
-        //update movements
-    }
+
+public interface CarService {
+
+    long deleteCarByName(String name);
```
Comment (rmycek) [inline src/main/java/com/guysfromusa/carsgame/services/CarService.java]: return the same type Long or long from all method

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class CarResource {
+
+    private final CarService carService;
+
+    @Inject
+    public CarResource(CarService carService) {
+        this.carService = carService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET, path = "/get-all")
```
Comment (rmycek) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: 1) just use default /v1/cars
2) use @GetMapping annotation

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public List<Car> getAllCars(){
+        List<CarEntity> carEntities = this.carService.loadAllCars();
+        return CarConverter.toCars(carEntities);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, path = "{name}/new/{type}/car")
```
Comment (rmycek) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: path should be /v1/cars 
and change @RequestMapping to @PostMapping

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public Long addCar(@PathVariable("name") String name, @PathVariable("type") String type){
+        CarType carType = CarType.valueOf(type);
+
+        return carService.addCar(carType, name);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, path= "{name}/delete")
```
Comment (rmycek) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: the same 
path = /v1/cars
method DELETE
change to DeleteMapping

File: src/main/java/com/guysfromusa/carsgame/entities/enums/CarType.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.guysfromusa.carsgame.entities.enums;
+
+public enum CarType {
+
+    NORMAL(1),
+    MONSTER(1),
+    RACER(2);
+
+    private Integer maxStepsPerRequest;
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/entities/enums/CarType.java]: int

File: src/main/java/com/guysfromusa/carsgame/services/CarService.java [hunk 6]
```diff
@@ -1,55 +1,21 @@
...
-        Function<Direction, Direction> turnF = turnSide == LEFT ? Direction::turnLeft : Direction::turnRight;
-
-        Direction newDirection = turnF.apply(car.getDirection());
-        car.setDirection(newDirection);
-        */
-
-        //update movements
-    }
+
+public interface CarService {
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/services/CarService.java]: good point, proxy wont work without interface

File: src/main/java/com/guysfromusa/carsgame/services/impl/CarServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        this.carRepository = carRepository;
+    }
+
+    @Override
+    public long deleteCarByName(String name) {
+        return carRepository.deleteByName(name);
+    }
+
+    @Override
+    public Long addCar(CarType carType, String name) {
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/services/impl/CarServiceImpl.java]: lets use small longs always when we are sure that there cant be null, then we will know (without checking) null defence checks are not needed

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+@RestController
+@RequestMapping(value= "/v1/cars", produces =  APPLICATION_JSON_UTF8_VALUE)
+public class CarResource {
+
+    private final CarService carService;
+
+    @Inject
+    public CarResource(CarService carService) {
+        this.carService = carService;
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: notNull(carService) to fail fast during spring initialization (once bean was not injected -> forgotten @Component or sth)

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java (hunk 2 above)
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: GET /cars  should return all cars

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java (hunk 3 above)
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: /cars POST should accept JSON body with car details needed to create cars

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java (hunk 4 above)
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: DELETE /cars/{name} - no verbs

File: src/main/java/com/guysfromusa/carsgame/v1/converters/CarConverter.java [hunk 9]
```diff
@@ -18,6 +19,14 @@
     public static Car toCar(CarEntity carEntity){
         Car car = new Car();
         car.setName(carEntity.getName());
+        car.setType(carEntity.getCarType().name());
+
+        Integer positionX = carEntity.getPositionX();
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/converters/CarConverter.java]: try to update test before implementation

File: src/main/java/com/guysfromusa/carsgame/v1/model/Car.java [hunk 10]
```diff
@@ -21,4 +20,7 @@
     @Getter @Setter
     private Direction direction;
 
+    @Getter @Setter
+    private String type;
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/model/Car.java]: enum?

File: src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = SpringContextConfiguration.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```
Comment (tbradlo) [inline src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java]: do we need dirties context?

**Pull request merged**